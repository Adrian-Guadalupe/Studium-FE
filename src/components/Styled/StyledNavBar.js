import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StyledAvatar from './StyledAvatar';
import { ReactComponent as SmallWhiteLogo } from '../../images/SmallWhiteLogo.svg';
import { Popover } from 'antd';
import { signout } from '../../actions';
import SmallLogo from './SmallLogo';
import { useDispatch, useSelector } from 'react-redux';
import StyledContainer from './StyledContainer';

const StyledNavBar = ( { visable, ...props } ) => {
  
  const user = useSelector( state => state.usersReducer );
  const [ menuOpen, setMenuOpen ] = useState( false );
  const [ avatarUrl, setAvatarUrl ] = useState( false );
  const dispatch = useDispatch();
  
  useEffect( () => {
    if( visable ){
      if( user.user && user.user.photoURL ){
        setAvatarUrl( user.user.photoURL );
      }else{
        setAvatarUrl( false );
      }
    }else{
      setAvatarUrl( false );
    }
  }, [ visable, user ] );
  
  const onAvatarClick = () => {
    setMenuOpen( !menuOpen );
  };
  
  const logout = () => {
    setMenuOpen( false );
    signout( dispatch );
  };
  
  const content = ( <div>
    <StyledLink onClick={ logout }>Logout</StyledLink>
  </div> );
  
  return ( <StyledBar visable={ visable }>
    <StyledContainer justifyContent={ 'space-between' }>
      <SmallWhiteLogo style={ {
        position: 'absolute', left: '6%', top: '45%',
      } }/>
      <Popover content={ content } visible={ menuOpen }
               placement="bottomRight">
        <StyledAvatar onClick={ onAvatarClick } avatarUrl={ avatarUrl }
                      className={ 'ant-dropdown-link' }/>
      </Popover>
    </StyledContainer>
  </StyledBar> );
};

StyledNavBar.propTypes = {};

const StyledBar = styled.div`
background-color: #585858;
display: flex;
justify-content: center;
z-index: 15;
position: absolute;
top: ${ props => props.visable ? '0' : '-74px' };
width: 100%;
height: 74px;
transition: all 2s;

`;

const StyledLink = styled.a`
font-weight: bold;
`;

export default StyledNavBar;