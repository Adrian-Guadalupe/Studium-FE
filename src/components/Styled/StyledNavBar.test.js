import React from 'react';
import renderer from 'react-test-renderer';

import StyledNavBar from './StyledNavBar';

test('StyledNavBar render correctly', () => {
    const tree = renderer.create(<StyledNavBar />).toJSON();
    expect(tree).toMatchSnapshot();
});