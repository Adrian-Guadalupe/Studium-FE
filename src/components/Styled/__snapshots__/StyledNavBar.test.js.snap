// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledNavBar render correctly 1`] = `
.c1 {
  box-sizing: border-box;
  top: 0;
  margin: 0;
  position: relative;
  width: 100%;
  height: 75px;
  max-height: 100%;
  max-width: 700px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow-y: scroll;
}

.c1::-webkit-scrollbar {
  width: 0;
}

.c0 {
  background-color: #585858;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 15;
  position: absolute;
  top: -74px;
  width: 100%;
  height: 74px;
}

@media screen and (min-width:768px) {
  .c0 {
    width: 400px;
    height: 100vh;
    left: 0;
    top: 0;
  }
}

<div>
  <div
    class="c0 nav-bar"
  >
    <div
      class="c1 nav-bar-container"
      height="75px"
    >
      <svg
        style="position: absolute; left: 6%; top: 50%; transform: transition(0, -53%);"
      >
        SmallWhiteLogo.svg
      </svg>
      <svg
        class="ant-dropdown-link"
        style="position: absolute; top: 15px; right: 8%;"
      >
        ProfileIcons.svg
      </svg>
    </div>
  </div>
</div>
`;
